### PR TITLE
ticket 7 completed

### PR DESCRIPTION
**Pull request by chloewong108**

### CODE REVIEW

File: src/api.js [hunk 1]
```diff
@@ -31,3 +31,15 @@ export const getArticleById = (article_id) => {
...
+    `https://bc-news-example.herokuapp.com/api/articles/${article_id}`,
+    { inc_votes: 1 }
+  );
+};
+export const reduceVotes = (article_id) => {
+  return axios.patch(
+    `https://bc-news-example.herokuapp.com/api/articles/${article_id}`,
+    { inc_votes: -1 }
+  );
+};
```
Comment (douglashellowell) [inline src/api.js]: these look very similar 🕵️‍♀️

could you make this more DRY?

File: src/components/home.jsx [hunk 2]
```diff
@@ -12,7 +12,7 @@ const Home = () => {
       setIsLoading(false);
     });
   }, []);
-  if (isLoading) return <p>Loading...</p>;
+  if (isLoading) return <p>🐢 Loading...🐢</p>;
```
Comment (douglashellowell) [inline src/components/home.jsx]: love the emojis!

whereever you use them be sure to consider accessibility

best practice is to wrap them in a `<span>` with some appropriate tags

> if you want them to be announced by screen readers
```jsx
  if (isLoading) return <p><span role="img" aria-label="loading-turtle">🐢</span> Loading...<span role="img" aria-label="loading-turtle">🐢</span></p>;
```

> if you want them to be ignored (just read out "loading...")
```jsx
  if (isLoading) return <p><span role="img" aria-hidden={true}>🐢</span> Loading...<span role="img" aria-hidden={true}>🐢</span></p>;
```

File: src/components/articleCard.jsx [hunk 3]
```diff
@@ -21,16 +23,47 @@ const ArticlePage = () => {
...
+            setVotes((currVotes) => {
+              return currVotes - 1;
+            });
+          });
+          setVotes((currVotes) => {
+            return currVotes + 1;
+          });
+        }}
+      >
+        👍
```
Comment (douglashellowell) [inline src/components/articleCard.jsx]: same goes with these emojis!

some help with A11Y needed here

**Comment by douglashellowell** [comment]
Great work!

At the moment users can't change their minds if they vote - which is okay if you have designed it to be that way!
If you want you users to be able to change their minds you will need to refactor some of your logic here.

---

Functionally it's working great but there are some A11Y and UX considerations to make

Along with the emoji comments i made above you should also think about how to visually show that the buttons can't be clicked anymore - at the moment they are disabled but they still _look_ like they should be clickable!

---

please put in another `/nchelp pr` when you've made these changes!

**Review by emilyb93 — APPROVED** [review]
Great changes :D

**Pull request merged**